garage/tf/algos/off_policy_batch_polopt.py

```diff
@@ -0,0 +1,179 @@
+                        samples_data["success_history"])
+                    self.log_diagnostics(paths)
+                    for train_itr in range(self.n_train_steps):
+                        if (self.replay_buffer.n_transitions_stored >=
+                                self.min_buffer_size) or (
+                                    self.replay_buffer_type == Buffer.HER):
+                            self.evaluate = True
+                            qf_loss, y, q, policy_loss = self.optimize_policy(
+                                epoch, samples_data)
+
+                            episode_policy_losses.append(policy_loss)
+                            episode_qf_losses.append(qf_loss)
+                            epoch_ys.append(y)
+                            epoch_qs.append(q)
+
+                    if self.plot:
+                        self.plotter.update_plot(self.policy,
+                                                 self.max_path_length)
+                        if self.pause_for_plot:
+                            input("Plotting evaluation run: Press Enter to "
+                                  "continue...")
+
+                logger.log("Training finished")
+                logger.log("Saving snapshot #{}".format(epoch))
+                params = self.get_itr_snapshot(epoch, samples_data)
+                logger.save_itr_params(epoch, params)
+                logger.log("Saved")
+                if self.evaluate:
+                    logger.record_tabular('Epoch', epoch)
+                    logger.record_tabular('AverageReturn',
+                                          np.mean(episode_rewards))
+                    logger.record_tabular('StdReturn', np.std(episode_rewards))
+                    logger.record_tabular('Policy/AveragePolicyLoss',
+                                          np.mean(episode_policy_losses))
+                    logger.record_tabular('QFunction/AverageQFunctionLoss',
+                                          np.mean(episode_qf_losses))
+                    logger.record_tabular('QFunction/AverageQ',
+                                          np.mean(epoch_qs))
+                    logger.record_tabular('QFunction/MaxQ', np.max(epoch_qs))
+                    logger.record_tabular('QFunction/AverageAbsQ',
+                                          np.mean(np.abs(epoch_qs)))
+                    logger.record_tabular('QFunction/AverageY',
+                                          np.mean(epoch_ys))
+                    logger.record_tabular('QFunction/MaxY', np.max(epoch_ys))
+                    logger.record_tabular('QFunction/AverageAbsY',
+                                          np.mean(np.abs(epoch_ys)))
+                    if self.replay_buffer_type == Buffer.HER:
+                        logger.record_tabular('AverageSuccessRate',
+                                              np.mean(self.success_history))
+
+                # Uncomment the following if you want to calculate the average
```

Comment:
Please find a different way of doing this or remove it--commenting/uncommenting code is not a valid way of changing a module's behavior.